crates/e2e-test-utils/src/traits.rs

```diff
@@ -0,0 +1,22 @@
+use reth::rpc::types::{
+    engine::{ExecutionPayloadEnvelopeV3, OptimismExecutionPayloadEnvelopeV3},
+    ExecutionPayloadV3,
+};
+
+/// The execution payload envelope type.
+pub trait PayloadEnvelopeV3: Send + Sync + std::fmt::Debug {
```

Comment:
we need to extend this in the future so I'd like to rename this to `EthPayloadEnvelopeExt`